src/Silverback.Integration.Kafka.Testing/Messaging/Broker/Kafka/Mocks/InMemoryPartition.cs

```diff
@@ -48,12 +50,26 @@ public Offset Add(Message<byte[]?, byte[]?> message)
             }
         }
 
-        public bool TryPull(Offset offset, out ConsumeResult<byte[]?, byte[]?>? result)
+        public bool TryPull(Offset offset, bool enablePartitionEof, out ConsumeResult<byte[]?, byte[]?>? result)
         {
             lock (_messages)
             {
                 if (offset.Value > LastOffset || _messages.Count == 0)
                 {
+                    if (enablePartitionEof && (_lastReportedOffsetForEndOfPartition < offset.Value && _messages.Count > _lastReportedOffsetForEndOfPartition ||
+                                               _lastReportedOffsetForEndOfPartition == null && _messages.Count == 0))
```

Comment:
I assume you mean the condition _messages.Count > _lastReportedOffsetForEndOfPartition: Because we don't increase the offset in case of end of partition, we need to make sure it's not reporting end of partition all the time.

Reply:
Isn't that ensured by `_lastReportedOffsetForEndOfPartition < offset.Value`? As soon as you report it once, `_lastReportedOffsetForEndOfPartition` should be incremented, no? `_lastReportedOffsetForEndOfPartition = offset.Value` -> therefore is not `< offset.Value` anymore. What am I missing?

Reply:
You are right, it's not needed!